src/Functions/JSONPath/Parsers/ParserJSONPathMemberAccess.cpp

```diff
@@ -0,0 +1,42 @@
+#include <Functions/JSONPath/Parsers/ParserJSONPathMemberAccess.h>
+#include <Functions/JSONPath/ASTs/ASTJSONPathMemberAccess.h>
+
+#include <Parsers/Lexer.h>
+#include <Parsers/ExpressionElementParsers.h>
+#include <Parsers/ASTIdentifier.h>
+
+namespace DB
+{
+
+/**
+ *
+ * @param pos token iterator
+ * @param node node of ASTJSONPathMemberAccess
+ * @param expected stuff for logging
+ * @return was parse successful
+ */
+bool ParserJSONPathMemberAccess::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
+{
+    auto member_access = std::make_shared<ASTJSONPathMemberAccess>();
+    node = member_access;
+    if (pos->type != TokenType::Dot) {
+        return false;
+    }
```

Comment:
We made it a rule to always initialize AST node passed to us in the very beginning of parseImpl to not get confused, since doing so is less error-prone; and it's a quite small allocation by the way

Reply:
Hmmm, actually okay, I agree with you. Will correct it in next commit

Reply:
done